src/lib.nix

```diff
@@ -69,26 +69,29 @@
     forAllSystems = f: lib.mapAttrs f allPkgs;
 
     dream2nixFor = forAllSystems (dream2nixForSystem config);
-  in {
-    riseAndShine = throw "Use makeFlakeOutputs instead of riseAndShine.";
-
-    makeFlakeOutputs = mArgs:
-      makeFlakeOutputsFunc
-      (
-        {inherit config pkgs systems;}
-        // mArgs
-      );
-
-    apps =
-      forAllSystems
-      (system: pkgs:
-        dream2nixFor."${system}".apps.flakeApps);
-
-    defaultApp =
-      forAllSystems
-      (system: pkgs:
-        dream2nixFor."${system}".apps.flakeApps.dream2nix);
-  };
+  in
+    if pkgs != null
+    then dream2nixFor."${makePkgsKey pkgs}"
+    else {
```

Comment:
I think it is confusing if init results in different structure, depending on the input.
Maybe `makeFlakeOutputs` should be removed entirely, or if kept, moved to the same level as `makeOutputs`.
Implementation wise I think makeFlakeOutputs should just be a wrapper around `makeOutputs`.

Reply:
One argument for keeping makeFlakeOutputs is, that it allows us to have a very very simple example flake in the readme.
Also, having the kapability to quickly generate flake structure can come in handy for future CLI tools, where one wants to quickly `flakify` some arbitrary project via one command.

Reply:
`makeFlakeOutputs` being on the same level as `makeOutputs` doesn't really make sense because `makeOutputs` is in `src/default.nix`, which requires a nixpkgs pkgs set. I think we should change `makeFlakeOutputs` so that it initializes the dream2nix instance(s) and outputs in one call (instead of needing to call init and then makeFlakeOutputs). This would make it so that `init` takes only a pkgs (or one system too maybe, for ease of use). I think this is better than the current `makeFlakeOutputs` usage; it already "takes over" the flake outputs, but you need to go to the trouble of initializing dream2nix and then using that, which doesn't make sense to me if it already takes over the outputs (meaning you wont use the instance again, only once), you should just be able to do it in one call.

Reply:
OK, makes sense. I'm fine with all of what you just said.

Reply:
I pushed these changes to the PR just now.